GLAA.Repository/EntityFrameworkRepositoryBase.cs

```diff
@@ -3,37 +3,55 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Linq.Expressions;
+using GLAA.Common;
 
 namespace GLAA.Repository
 {
     public class EntityFrameworkRepositoryBase : IEntityFrameworkRepository
     {
-        protected readonly GLAAContext Context;        
+        protected readonly GLAAContext Context;
+        protected readonly IDateTimeProvider DateTimeProvider;
 
-        public EntityFrameworkRepositoryBase(GLAAContext context)
+        public EntityFrameworkRepositoryBase(GLAAContext context, IDateTimeProvider dtp)
         {
-            this.Context = context;              
+            Context = context;
+            DateTimeProvider = dtp;
         }
 
-        public TEntity GetById<TEntity>(int id) where TEntity : class, IId
+        public TEntity GetById<TEntity>(int id, bool includeDeleted = false) where TEntity : class, IId
```

Comment:
I think I'd rather have separate methods for getting live and non live records as I can forsee reasons for them diverging over time. Also not a massive fan of methods which can perform separate roles via parameters, happy to leave for the time being but one to discuss. 

If they do diverge it may be worth us having a separate repo even, rather than methods for handling that, with a generic type constraint on the `IDeletable` interface

Reply:
Good points. I think this parameter way of working makes sense for getting collections of records. I'll split out the methods getting individual records.

Reply:
Yeah an admin repo may be one route, but we should consider that on some pages it may be that we require different data depending on the user (but in the same  service/controller class) so we'll need to decide in which layer we make the decision about what data to return, service, repo, method in repo etc